src/app/store/networks/networks.slice.ts

```diff
@@ -0,0 +1,67 @@
+import { createEntityAdapter, createSlice, EntityId, PayloadAction } from '@reduxjs/toolkit';
+import { ChainID } from '@stacks/transactions';
+
+import { DEFAULT_MAINNET_SERVER, DEFAULT_TESTNET_SERVER } from '@shared/constants';
+
+export type NetworkModes = 'mainnet' | 'testnet' | 'devnet';
+
+export enum DefaultNetworkIds {
+  mainnet = 'mainnet',
+  testnet = 'testnet',
+  devnet = 'devnet',
+}
+
+export interface Network {
+  chainId: ChainID;
+  id: string;
+  name: string;
+  url: string;
+}
+
+const defaultCurrentNetworkId = DefaultNetworkIds.mainnet as EntityId;
+
+export const defaultCurrentNetwork = {
+  chainId: ChainID.Mainnet,
+  id: DefaultNetworkIds.mainnet,
+  name: 'Mainnet',
+  url: DEFAULT_MAINNET_SERVER,
+} as const;
+
+export const defaultNetworks = [
+  defaultCurrentNetwork,
+  {
+    chainId: ChainID.Testnet,
+    id: DefaultNetworkIds.testnet,
+    name: 'Testnet',
+    url: DEFAULT_TESTNET_SERVER,
+  },
+  {
+    chainId: ChainID.Testnet,
+    id: DefaultNetworkIds.devnet,
+    name: 'Devnet',
+    url: 'http://localhost:3999',
+  },
+] as const;
```

Comment:
Think it was tider keying the defaults by id as before (would also help spreading onto the selector too)


```ts
export const defaultNetworks: Networks = {
  mainnet: {
    url: DEFAULT_MAINNET_SERVER,
    name: 'Mainnet',
    chainId: ChainID.Mainnet,
  },
  testnet: {
    url: DEFAULT_TESTNET_SERVER,
    name: 'Testnet',
    chainId: ChainID.Testnet,
  },
  devnet: {
    url: 'http://localhost:3999',
    name: 'Devnet',
    chainId: ChainID.Testnet,
  },
} as const;
```

Reply:
You can do either with `createEntityAdapter`, happy to change if easier for you to read.

Reply:
I feel like leaving it as an array of Networks makes more sense to me rather than having a keyed object with a prop `id` too. `createEntityAdapter` is setup to use the id prop provided to create the keyed dictionary for you so as long as the id is included, the array makes sense here? Then when you use `networkSelectors.selectEntities` it just returns a `Dictionary<Network>`.

Reply:
Whether we store the hard coded networks as an array or dictionary I don't really mind.

The concern is only what we ask the browser to save. Persisting data on the client introduces a whole host of new things to consider when making changes later etc. We want to persist the absolute minimum data possible. Hence the ask to persist only user-added networks, and expose the defaults via a selector.

Reply:
I get it now! Haha. Thanks.